src/dash/models/DashManifestModel.js

```diff
@@ -589,6 +589,50 @@ function DashManifestModel() {
         });
     }
 
+    // propertyType is one of { DashConstants.ESSENTIAL_PROPERTY, DashConstants.SUPPLEMENTAL_PROPERTY }
+    function _getCommonPropertiesFromRepresentations(propertyType, repr) {
```

Comment:
I suggest to use a `Set` to get O(1) lookup times:

````js
    function _getCommonPropertiesFromRepresentations(propertyType, repr) {
        if (!repr || !repr.length) {
            return [];
        }

        let propertiesOfFirstRepresentation = repr[0][propertyType] || [];

        if (propertiesOfFirstRepresentation.length === 0) {
            return [];
        }

        if (repr.length === 1) {
            return propertiesOfFirstRepresentation;
        }

        const propSet = new Set(repr.slice(1).flatMap(currRep =>
            currRep[propertyType]?.map(e => `${e.schemeIdUri}-${e.value}`) || []
        ));

        return propertiesOfFirstRepresentation.filter(prop =>
            propSet.has(`${prop.schemeIdUri}-${prop.value}`)
        );
    }
````

Reply:
Uh, I think, `new Set(flatMap())` does not solve our problem here:
Doesn't it combine all possible Properties from [1..n] instead of selecting those common to all?

Reply:
I think you are right, it's good that you added a third Representation in the unit tests to catch my mistake. Let's keep your original code.